### DISCUSSION TITLE
Virtual layers: not working through QFieldCloud but ok otherwise. Any solution?

### DISCUSSION THREAD

**Discussion by jasoncl-5w** [opengisch/QField#5106] [Q&A]
Hello everyone,
I'm trying to use a virtual layer for a QField project. It works perfectly in QGIS, and it seems to work as well when packaging the project for QField before importing it on the Android phone from local files. However, when using QFieldCloud, which I have to, the use of a virtual layer lead to an error so that the download is not possible. The error is wrong though: it says "(Downloading error. [QF/permission_denied_plan_inssufficient] Permission denied because the user acount's plan is insufficient)", while I can guarantee the available space is largely enough. 
Has anyone an idea how to work, if possible, properly with virtual layers through QFieldCloud?
Thanks,
Jason

**Comment by elmonchas** [comment]
Same problem here with a virtual layer in a project for qfield cloud. 

Anybody any idea how to solve this error?

**Comment by nirvn** [comment]
@jasoncl-5w , could you share a sample of your project that fails?

**Comment by elmonchas** [comment]
@nirvn Here are some pointers from my case. In the project overview on cloud it gives this warning: 
![image](https://github.com/user-attachments/assets/a86a9111-60a0-4f90-a113-4abd169fcee7)

When I check the log I see this: 
![image](https://github.com/user-attachments/assets/6eb63c4a-5d46-4a78-9073-362f17baf46c)

When I re-open the project on qgis desktop. The virtual layer also have an unknown datasource
![image](https://github.com/user-attachments/assets/2c474d50-1643-4f4d-a37c-b91648590b2b)

The option "no geometry" was selected. I als otried automatic handling, but same result. 
This is the probably faulty query:

-- Step 1: Create an aggregated table from invoer_2025
WITH nieuw_tabel AS (
    SELECT
        soortlijst2025_wet_naam AS wet_naam,
        COUNT(distinct hok_id) AS N_hokken_nieuw
    FROM invoer_snl_2025
    GROUP BY soortlijst2025_wet_naam
),

-- Step 2: Create an aggregated table from oude_stippen
oud_tabel AS (
    SELECT
        soort_wet AS wet_naam,
        COUNT(distinct hok_id) AS N_hokken_oud
    FROM oude_stippen_SNLonly
    GROUP BY wet_naam
)

-- Step 3: Join both aggregated tables on wet_naam
SELECT 
    subquery.wet_naam, 
    subquery.N_hokken_nieuw, 
    subquery.N_hokken_oud
FROM (
    SELECT
        COALESCE(n.wet_naam, o.wet_naam) AS wet_naam,  -- Keep non-matching values
        COALESCE(n.N_hokken_nieuw, 0) AS N_hokken_nieuw,  -- Replace NULL with 0 for missing values in nieuw_tabel
        COALESCE(o.N_hokken_oud, 0) AS N_hokken_oud  -- Replace NULL with 0 for missing values in oud_tabel
    FROM nieuw_tabel n
    FULL OUTER JOIN oud_tabel o ON n.wet_naam = o.wet_naam
) AS subquery;

**Comment by nirvn** [comment]
@elmonchas , please provide a test project

**Comment by elmonchas** [comment]


**Reply by elmonchas** [reply]
[sample project.zip](https://github.com/user-attachments/files/19546245/sample.project.zip)

**Reply by SeqLaz** [reply]
Hi @elmonchas ,

After reviewing your project, I noticed that your project has virtual layers, which are currently not available in the Community plan of QFieldCloud. However, it is in our backlog, and we plan to make it available for all plans in the future.

In the meantime, the recommendation is to save these virtual layers that you created with SQL and save them as a GeoPackage.

**Reply by elmonchas** [reply]
virtual layer support in all plans would be great!  The issue with the virtual layer actually happens with a paid plan. This seemed to be the case for @jasoncl-5w too.

Anyway this might be solved with the upcoming changes then
